src/app/painter.cljc

```diff
@@ -12,7 +12,8 @@
             [clojure.string :as str]
             [missionary.core :as m]))
 
-#?(:clj (defonce canvas-state (atom [])))
+#?(:clj (defonce !canvas-state (atom [])))
+(e/def canvas-state (e/server (e/watch !canvas-state)))
```

Comment:
try to watch each atom once, otherwise you can introduce inconsistent renders (the watches are not synchronized)